src/components/gitOps/GitOpsConfiguration.tsx

```diff
@@ -190,267 +201,536 @@ class GitOpsConfiguration extends Component<GitOpsProps, GitOpsState> {
             },
             isError: {
                 ...this.state.isError,
-                [key]: event.target.value.length === 0 ? "This is a required field" : "",
+                [key]: event.target.value.length === 0 ? 'This is a required field' : '',
             },
             isFormEdited: false,
+            //After entering any text,if GitOpsFieldKeyType is of type host then the url validation error must dissapear
+            isUrlValidationError: key === 'host' ? false : this.state.isUrlValidationError,
         })
     }
 
     getFormErrors(isFormEdited, form: GitOpsConfig): any {
-        if (!isFormEdited) return {
-            host: "",
-            username: "",
-            token: "",
-            gitHubOrgId: "",
-            gitLabGroupId: "",
-            azureProjectName: "",
-            bitBucketWorkspaceId: "",
-            bitBucketProjectKey: ""
-        }
+        if (!isFormEdited)
+            return {
+                host: '',
+                username: '',
+                token: '',
+                gitHubOrgId: '',
+                gitLabGroupId: '',
+                azureProjectName: '',
+                bitBucketWorkspaceId: '',
+                bitBucketProjectKey: '',
+            }
 
         let isError = {
-            host: form.host.length ? "" : "This is a required field",
-            username: form.username.length ? "" : "This is a required field",
-            token: form.token.length ? "" : "This is a required field",
-            gitHubOrgId: form.gitHubOrgId.length ? "" : "This is a required field",
-            gitLabGroupId: form.gitLabGroupId.length ? "" : "This is a required field",
-            azureProjectName: form.azureProjectName.length ? "" : "This is a required field",
-            bitBucketWorkspaceId: form.bitBucketWorkspaceId.length ? "" : "This is a required field",
-            bitBucketProjectKey: ""
-        };
-        return isError;
+            host: form.host.length ? '' : 'This is a required field',
+            username: form.username.length ? '' : 'This is a required field',
+            token: form.token.length ? '' : 'This is a required field',
+            gitHubOrgId: form.gitHubOrgId.length ? '' : 'This is a required field',
+            gitLabGroupId: form.gitLabGroupId.length ? '' : 'This is a required field',
+            azureProjectName: form.azureProjectName.length ? '' : 'This is a required field',
+            bitBucketWorkspaceId: form.bitBucketWorkspaceId.length ? '' : 'This is a required field',
+            bitBucketProjectKey: '',
+        }
+        return isError
     }
 
     isInvalid() {
```

Comment:
this method combined with other method is convoluted logic

Reply:
done